pkgs/applications/networking/instant-messengers/gtkcord/default.nix

```
@@ -0,0 +1,45 @@
+{ makeDesktopItem, lib, buildGoModule, fetchFromGitHub, pkgconfig, go, gnome3, libhandy }:
+
+buildGoModule rec {
+  pname = "gtkcord";
+  version = "0.0.4";
+
+  src = fetchFromGitHub {
+    owner = "diamondburned";
+    repo = "gtkcord3";
+    rev = "v${version}";
+    sha256 = "0n9600s50zpx8kv89icrlxarm3633gnizg1sq1zpask8bbv2xnfd";
+  };
+
+  nativeBuildInputs = [ pkgconfig go ];
+  
+  buildInputs = [ gnome3.glib gnome3.gtk libhandy ];
+
+  desktopFile = makeDesktopItem {
```

Also, ideally upstream would create the desktop file and install it.

---

Meaning I shouldn't be creating the desktop file in the nix expression?

---

It is fine until there is one upstream but the point of desktop files is that software projects can include them so that each distribution does not have to create their own launcher for each desktop environment. Opening an issue in gtkcord tracker about this and adding a comment with a link would be appreciated.

---

Ah wasn't quite sure what you meant by "upstream". Submitted an [issue](https://github.com/diamondburned/gtkcord3/issues/77)

---

Could you also please add a comment linking to the issue to the line above, so it is easier to check if it is resolved when updating the package?

---

Could you also please add a comment linking to the issue to the line above, so it is easier to check if it is resolved when updating the package?

---

Updated the issue body so that it also links to the specific nix expression line containing the `desktopFile` attribute :+1:

---

@skykanin I think @jtojnar meant:

```suggestion
  # No upstream desktop file yet: https://github.com/diamondburned/gtkcord3/issues/77
  desktopFile = makeDesktopItem {
```

---

Oh I misunderstood, fixed